test/zdtm/static/cgroup_yard.hook

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+
+import sys
+import os
+import subprocess
+import tempfile
+
+yard = "external_yard"
+
+if sys.argv[1] == "--pre-dump":
+	os.mkdir(yard)
+	with open("/proc/self/cgroup") as f:
+		for line in f:
+			cgr = line.split(":")[1]
+
+			if cgr == "":
+				continue
+
+			if cgr.startswith("name="):
+				ctrl = cgr[len("name="):]
+				opts = "none," + cgr
+			else:
+				ctrl = cgr
+				opts = cgr
+
+			os.mkdir(yard + "/" + ctrl)
+			subprocess.call(["mount", "-t", "cgroup", "none", yard + "/" + ctrl, "-o", opts])
+
+if sys.argv[1] == "--post-restore":
+	with open("/proc/self/cgroup") as f:
+		for line in f:
+			cgr = line.split(":")[1]
+
+			if cgr == "":
+				continue
+
+			if cgr.startswith("name="):
+				ctrl = cgr[len("name="):]
+			else:
+				ctrl = cgr
+
+			subprocess.call(["umount", yard + "/" + ctrl])
+			os.rmdir(yard + "/" + ctrl)
+	os.rmdir(yard)
+
+if sys.argv[1] in ["--clean", "--pre-restore"]:
+	tname = tempfile.mkdtemp()
+	subprocess.call(["mount", "-t", "cgroup", "none", tname, "-o", "none,name=zdtmtst"])
+
+	try:
+		os.rmdir(os.path.join(tname, "subcg00", "subsubcg"))
+	except:
+		pass
+
+	try:
+		os.rmdir(os.path.join(tname, "subcg00"))
+	except:
+		pass
+
+	subprocess.call(["umount", tname])
+	os.rmdir(tname)
```

Comment:
Can we replace `os.rmdir` with something like `shutil.rmtree(tname, ignore_errors=True)`?

Reply:
I think we can but `shutil.rmtree` will try to remove the files in that directory first, fail on all of them and ignore that errors and then remove the directory. I'm not sure if it's the cleanest solution.

Reply:
if os.accesss(path, os.F_OK):
  os.rmdir(path)